### PR TITLE
Replace callbacks with channels, refactoring

### PR DESCRIPTION
**Pull request by Jaskaranbir**
This makes this package much closer to Sarama's, except this still does some convenient bootstrapping.

### CODE REVIEW

File: consumer/consumer.go [hunk 1]
```diff
@@ -47,13 +44,12 @@ type Consumer struct {
 	consumer         Adapter
 	isClosed         bool
 	isLoggingEnabled bool
+	closeChan        chan bool
```
Comment (golangci) [inline consumer/consumer.go]: `closeChan` is unused

File: consumer/consumer_test.go [hunk 2]
```diff
@@ -120,44 +103,37 @@ var _ = Describe("Consumer", func() {
 			fakeConsumer.Close()
 		})
 
-		It("should run the error-handler function", func() {
+		It("should forward errors to error-channel", func() {
+			isErrorForwarded := false
 			var wg sync.WaitGroup
 			wg.Add(1)
+			go func() {
+				for _ = range fakeConsumer.Errors() {
```
Comment (golangci) [inline consumer/consumer_test.go]: should omit values from range; this loop is equivalent to `for range ...`

File: consumer/consumer_test.go [hunk 3]
```diff
@@ -166,84 +142,39 @@ var _ = Describe("Consumer", func() {
...
 
-		It("should run the message-handler function", func() {
+		It("should forward messages to message-channel", func() {
+			isMsgForwarded := false
 			var wg sync.WaitGroup
 			wg.Add(1)
 			go func() {
-				fakeConsumer.MessagesChan <- fakeConsumer.CreateMockMessage("test", "test", "test")
-				defer wg.Done()
+				for _ = range fakeConsumer.Messages() {
```
Comment (golangci) [inline consumer/consumer_test.go]: should omit values from range; this loop is equivalent to `for range ...`

File: producer/producer_test.go [hunk 4]
```diff
@@ -180,24 +164,25 @@ var _ = Describe("Producer", func() {
...
 			asyncProducer.Close()
 			mockBroker.Close()
 		})
 
 		It("should run the error-handler function", func() {
+			isErrorForwarded := false
 			var wg sync.WaitGroup
 			wg.Add(1)
+			go func() {
+				for _ = range fakeProducer.Errors() {
```
Comment (golangci) [inline producer/producer_test.go]: should omit values from range; this loop is equivalent to `for range ...`

**Comment by Jaskaranbir** [comment]
Closing PR. For some reason, its not being updated with new commits to the branch.

**Pull request closed without merging**